src/widgets/dock_area/mod.rs

```diff
@@ -348,6 +365,79 @@ impl<'tree, Tab> DockArea<'tree, Tab> {
                 }
             }
         }
+
+        separators
+    }
+
+    fn show_cross_section_separators(&mut self, ui: &mut Ui, separators: Vec<LineSeparator>) {
```

Comment:
I'd rename this to `show_separator_junctions`. We aren't really creating new separators here, just new interaction areas between existing ones. Also, a cross-section is the surface exposed when you slice through a solid object :)